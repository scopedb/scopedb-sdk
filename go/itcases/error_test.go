/*
 * Copyright 2024 ScopeDB, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package itcases

import (
	"context"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/google/uuid"
	"github.com/stretchr/testify/require"
)

func TestFetchStatementFail(t *testing.T) {
	c := NewClient(t)
	defer c.Close()

	ctx := context.Background()

	id, err := uuid.Parse("c8fe71d6-3695-11f0-85b3-063c3400fda9")
	require.NoError(t, err)
	err = c.StatementHandle(id).FetchOnce(ctx)
	require.Error(t, err)
	snaps.MatchSnapshot(t, err.Error())
}

func TestSubmitStatementFail(t *testing.T) {
	c := NewClient(t)
	defer c.Close()

	ctx := context.Background()

	_, err := c.Statement("SELECT UNKNOWN_FUNCTION()").Execute(ctx)
	require.Error(t, err)
	snaps.MatchSnapshot(t, err.Error())
}
