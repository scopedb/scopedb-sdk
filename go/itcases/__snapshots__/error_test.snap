
[TestStatementFailure - 1]
result set not found: c8fe71d6-3695-11f0-85b3-063c3400fda9
---

[TestStatementFailure - 2]
0: failed to prepare statement: "SELECT REGEXP_MATCH('some text', 'malformed regex: \\d')"
1: failed to parse input

========================================

error: failed to execute statement
 --> ScopeQL:1:34
  |
1 | SELECT REGEXP_MATCH('some text', 'malformed regex: \d')
  |        ^^^^^^^^^^^^              ^^^^^^^^^^^^^^^^^^^^^ invalid escape or unicode
  |        |
  |        while parsing expression
  |

---
